pkgs/top-level/perl-packages.nix

```diff
@@ -15501,6 +15501,31 @@ let self = _self // overrides; _self = with self; {
     };
   };
 
+  WWWYoutubeViewer = buildPerlPackage rec {
+    name = "WWW-YoutubeViewer-${version}";
+    version = "3.3.0";
+
+    src = fetchFromGitHub {
+      owner  = "trizen";
+      repo   = "youtube-viewer";
+      rev    = "${version}";
+      sha256 = "15xyrwv08fw8jmpydwzks26ipxnzliwddgyjcfqiaj0p7lwlhmx1";
+    };
+
+    propagatedBuildInputs = [
+      LWP
+      LWPProtocolHttps
+      DataDump
+      JSON
+    ];
+
+    meta = {
+      description = "A lightweight application for searching and streaming videos from YouTube";
```

Comment:
Sorry I didn't mention this earlier, but it would be nice to have a homepage attribute under meta as well:

    homepage = http://www.webupd8.org/2015/02/youtube-viewer-complete-youtube-client.html;

(Either that URL or the URL to the github page.)

Reply:
No problem. Took the github page because it seems newer. and rebased again.